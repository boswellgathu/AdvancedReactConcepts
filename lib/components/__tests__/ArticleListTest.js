import React from 'react';
// import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import ArticleList from '../ArticleList';

const testProps = {
  articles: {
    a: {id: 'a'},
    b: {id: 'b'}
  }
};

describe('ArticleList', () => {
  it('It renders correctly', () => {
    const wrapper =  shallow(
      <ArticleList 
        {...testProps}
      />
    );

    //expect(wrapper.node.props.children.length).toBe(2);
    expect(wrapper.find('Article').length).toBe(2);
    expect(wrapper).toMatchSnapshot();
  });
});