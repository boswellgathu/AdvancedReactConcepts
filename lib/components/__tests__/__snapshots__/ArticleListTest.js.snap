// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList It renders correctly 1`] = `
<div>
  <div
    style={
      Object {
        "borderBottomColor": "#aaa",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
    <div>
      Invalid Date
    </div>
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 15,
        }
      }
    >
      <a
        href={undefined}
      >
          
      </a>
    </div>
    <div
      style={
        Object {
          "paddingLeft": 20,
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "borderBottomColor": "#aaa",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
    <div>
      Invalid Date
    </div>
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 15,
        }
      }
    >
      <a
        href={undefined}
      >
          
      </a>
    </div>
    <div
      style={
        Object {
          "paddingLeft": 20,
        }
      }
    />
  </div>
</div>
`;
